### DISCUSSION TITLE
Unknown arg `expires_in` ... Did you mean `expires_at`?

### DISCUSSION THREAD

**Discussion by OllieJT** [nextauthjs/next-auth#6662] [Help] [answered]
I've been pulling my hair out over this for the past 24hrs. On what was a previously working setup (auth-js / sveltekit / prisma) I've been having issues logging in since. I Can't find others having thad the same issue.

I'm hoping this is just going to be a rubber duck session, but would love any help I can get as I'm running out of ideas to debug this.

I've gone so far as to:
 - drop all rows and re-push my db schema
 - go to my oauth accounts (google) and disconnect the app
 - try alternate oauth providers (patreon, discord)
 - I even rewrote the Prisma adapter, and made the `linkAccount()` method include `expires_at`:
 ```
 // ...
 linkAccount: async (data) => {
	const fixed_data = {...data, expires_at: data.expires_in };
	/// ...
},
 // ...
 ```

This is the recurring error I'm getting:
```
Unknown arg `expires_in` in data.expires_in for type AccountUncheckedCreateInput. Did you mean `expires_at`?
```

Here's the full log, with sensitive values redacted:

![Imgur](https://i.imgur.com/NfKTQew.png)

**Comment by OllieJT** [comment]
So the google adapters type seems to be `oidc`, but Patrons is `oauth` and seems to have a clearer error:

```js
[auth][error][AdapterError]: Read more at https://errors.authjs.dev#adaptererror
[auth][cause]: Error:
Invalid `prisma.account.create()` invocation:

{
  data: {
    provider: 'patreon',
    type: 'oauth',
    providerAccountId: '<VALUE_REMOVED>',
    access_token: '<VALUE_REMOVED>',
    expires_in: 2678400,
    ~~~~~~~~~~
    token_type: 'bearer',
    scope: 'identity identity[email]',
    refresh_token: '<VALUE_REMOVED>',
    version: '0.0.1',
    ~~~~~~~
    userId: '<VALUE_REMOVED>',
    expires_at: 2678400
  }
}

Unknown arg `expires_in` in data.expires_in for type AccountUncheckedCreateInput. Did you mean `expires_at`? Available args:
type AccountUncheckedCreateInput {
  id?: String
  userId: String
  type: String
  provider: String
  providerAccountId: String
  refresh_token?: String | Null
  access_token?: String | Null
  expires_at?: Int | Null
  token_type?: String | Null
  scope?: String | Null
  id_token?: String | Null
  session_state?: String | Null
}
Unknown arg `version` in data.version for type AccountUncheckedCreateInput. Available args:

type AccountUncheckedCreateInput {
  id?: String
  userId: String
  type: String
  provider: String
  providerAccountId: String
  refresh_token?: String | Null
  access_token?: String | Null
  expires_at?: Int | Null
  token_type?: String | Null
  scope?: String | Null
  id_token?: String | Null
  session_state?: String | Null
}


    at Yn.validate (/Users/olliejt/Developer/intentions/node_modules/@prisma/client/runtime/library.js:148:91)
    at Ur.createMessage (/Users/olliejt/Developer/intentions/node_modules/@prisma/client/runtime/library.js:163:13577)
    at /Users/olliejt/Developer/intentions/node_modules/@prisma/client/runtime/library.js:173:10007
    at runInChildSpan (/Users/olliejt/Developer/intentions/node_modules/@prisma/client/runtime/library.js:70:25749)
    at t._executeRequest (/Users/olliejt/Developer/intentions/node_modules/@prisma/client/runtime/library.js:173:9996)
```

**Comment by breaker05** [comment, marked as answer]
You will need to add those to your data model and database migration. I had this exact issue this week with adding the Azure Active Directory provider except Azure needed the `ext_expires_in` field.

Add these to your prisma Account model:

```
expires_in          Int?
version             String?
```

**Reply by OllieJT** [reply]
Thank you! I don't know why I didn't try this sooner... but it does seem like an anti-pattern.

I figured the source of truth was the schema defined by auth-js, and any discrepancies in naming would be handled by the adapter.

Though it seems this has solved my problem, I think there's a deeper issue here.
